x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java

```diff
@@ -164,193 +164,191 @@ protected void readInternal(ByteBuffer b) throws IOException {
 
         try {
             final CacheFile cacheFile = getCacheFileSafe();
-            try (Releasable ignored = cacheFile.fileLock()) {
```

Comment:
Same argument as in the other spot where I removed use of this method, what's the point of locking here when all we do is make the cache file be used after eviction has already been started, meaning all the writes going into it are wasted anyway?

Reply:
The point was to keep the file around in order to be able to process any concurrent read operation (if any) that would have required a gap to be filled. But there was no reason to prevent other listeners to be acquired nor an eviction to kick in. Your change improves all of that.